client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -323,12 +328,81 @@ def predict_with_id(
         return (id, _utils.body_to_json(response))
 
+    def batch_predict(
+            self,
+            df,
+            batch_size: int = 100,
+            compress: bool = False,
+            max_retries: int = http_session.DEFAULT_MAX_RETRIES,
+            retry_status: Set[int] = http_session.DEFAULT_STATUS_FORCELIST,
+            backoff_factor: float = http_session.DEFAULT_BACKOFF_FACTOR,
+            prediction_id: str = None,
+    ):
+        """
+        Makes a prediction using input `df`.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            A batch of inputs for the model.
+        compress : bool, default False
+            Whether to compress the request body.
+        batch_size : int, default 100
+            The number of rows to send in each request.
+        max_retries : int, default 13
+            Maximum number of retries on status codes listed in ``retry_status``.
+        retry_status : set, default {404, 429, 500, 503, 504}
+            Set of status codes, as integers, for which retry attempts should be made.  Overwrites default value.
+            Expand the set to include more. For example, to add status code 409 to the existing set, use:
+            ``retry_status={404, 429, 500, 503, 504, 409}``
+        backoff_factor : float, default 0.3
+            A backoff factor to apply between retry attempts.  Uses standard urllib3 sleep pattern:
+            ``{backoff factor} * (2 ** ({number of total retries} - 1))`` with a maximum sleep time between requests of
+            120 seconds.
+        prediction_id: str, default None
+            A custom string to use as the ID for the prediction request.  Defaults to a randomly generated UUID.
+        Returns
+        -------
+        prediction : pd.DataFrame
+            Output returned by the deployed model for `x`.
+
+        Raises
+        ------
+        RuntimeError
+            If the deployed model encounters an error while running the prediction.
+        requests.HTTPError
+            If the server encounters an error while handing the HTTP request.
+
+        """
+
+        pd = importer.maybe_dependency("pandas")
+        if pd is None:
+            raise ImportError("pandas is not installed; try `pip install pandas`")
+
+        # Set the retry config if it differs from current config.
+        self._session = http_session.set_retry_config(
+            self._session,
+            max_retries=max_retries,
+            status_forcelist=retry_status,
+            backoff_factor=backoff_factor,
+        )
+
+        # Split into batches
+        out_df_list = []
+        for i in range(0, len(df), batch_size):
+            batch = df[i:i+batch_size]
+            serialized_batch = batch.to_dict(orient="records")
+            response = self._predict(serialized_batch, self.batch_prediction_url(), compress, prediction_id)
+            out_df = pd.DataFrame.from_dict(response.json())
```

Comment:
Like we do in `predict_with_id()`:

```suggestion
            out_df = pd.DataFrame.from_dict(_utils.body_to_json(response))
```